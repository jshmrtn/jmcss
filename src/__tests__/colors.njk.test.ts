// eslint-disable-next-line @typescript-eslint/no-var-requires
const colorsTemplate = require("template/colors.njk");

type testColor = Omit<Color, "equals" | "toHex" | "toHsl" | "blend">;

describe("colors.njk", () => {
  test("single color without transparency gets rendered", () => {
    let renderedTemplate;
    try {
      const colors: testColor[] = [{ name: "test-color", r: 255, g: 255, b: 255, a: 1 }];
      renderedTemplate = colorsTemplate.render({
        colors,
      });
    } catch (error) {
      console.error(error);
    }
    expect(renderedTemplate).toMatchSnapshot();
    expect(renderedTemplate).toEqual("$test-color: rgb(255, 255, 255);\n");
  });

  test("multiple colors without transparency get rendered", () => {
    let renderedTemplate;
    try {
      const colors: testColor[] = [
        { name: "test-color", r: 255, g: 255, b: 255, a: 1 },
        { name: "test-color-two", r: 255, g: 255, b: 255, a: 1 },
      ];
      renderedTemplate = colorsTemplate.render({
        colors,
      });
    } catch (error) {
      console.error(error);
    }
    expect(renderedTemplate).toMatchSnapshot();
    expect(renderedTemplate).toEqual("$test-color: rgb(255, 255, 255);\n" + "$test-color-two: rgb(255, 255, 255);\n");
  });

  test("message gets rendered if no colors are present", () => {
    let renderedTemplate;
    try {
      const colors: testColor[] = [];
      renderedTemplate = colorsTemplate.render({
        colors,
      });
    } catch (error) {
      console.error(error);
    }
    expect(renderedTemplate).toMatchSnapshot();
    expect(renderedTemplate).toEqual("/* No colors defined */");
  });
});
