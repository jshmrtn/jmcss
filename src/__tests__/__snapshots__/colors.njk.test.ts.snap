// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`colors.njk message gets rendered if no colors are present 1`] = `"/* No colors defined */"`;

exports[`colors.njk multiple colors without transparency get rendered 1`] = `
"$test-color: rgb(255, 255, 255);
$test-color-two: rgb(255, 255, 255);
"
`;

exports[`colors.njk single color without transparency gets rendered 1`] = `
"$test-color: rgb(255, 255, 255);
"
`;
